test/test_autograd.py

```diff
@@ -4922,6 +4922,30 @@ def fn(*inputs):
             setattr(TestAutogradDeviceType, test_name, do_test)
 
 class TestAutogradComplex(TestCase):
+    @skipIfNoLapack
+    def test_complex_cholesky(self):
+        def func(x, upper):
+            x = 0.5 * (x + x.transpose(-1, -2).conj())
+            return torch.cholesky(x, upper)
+
+        def run_test(upper, dims):
+            x = torch.rand(*dims, requires_grad=True, dtype=torch.cdouble)
+            x = x + torch.eye(dims[-1])
+
+            gradcheck(func, [x, upper])
+            # TODO: gradgradcheck does not work
```

Comment:
`gradgradcheck` doesn't work because backward for `matmul` needs to be updated as well.

Reply:
It might be better to create a separate PR to update the backward for `mm`, `addmm`, and `matmul` I think. Let me know if you would like me to create a PR for that or if you would like to work on it (in which case feel free to add me as a reviewer).

Reply:
Alright, I was suspecting that, thanks. I will make the PR.